engine/order_manager.go

```diff
@@ -284,6 +284,61 @@ func (m *OrderManager) validate(newOrder *order.Submit) error {
 	return nil
 }
 
+// Modify depends on the order.Modify.ID and order.Modify.Exchange fields to uniquely
+// identify an order to modify.
+func (m *OrderManager) Modify(mod *order.Modify) (*order.ModifyResponse, error) {
+	if m == nil {
+		return nil, fmt.Errorf("order manager %w", ErrNilSubsystem)
+	}
+	if atomic.LoadInt32(&m.started) == 0 {
+		return nil, fmt.Errorf("order manager %w", ErrSubSystemNotStarted)
+	}
+
+	// Fetch details from locally managed order store.
+	det, err := m.orderStore.getByExchangeAndID(mod.Exchange, mod.ID)
+	if det == nil || err != nil {
+		return nil, fmt.Errorf("order does not exist: %w", err)
+	}
+
+	// Populate additional Modify fields as some of them are required by various
+	// exchange implementations.
+	mod.Pair = det.Pair                           // Used by Bithumb.
+	mod.Side = det.Side                           // Used by Bithumb.
+	mod.PostOnly = det.PostOnly                   // Used by Poloniex.
+	mod.ImmediateOrCancel = det.ImmediateOrCancel // Used by Poloniex.
```

Comment:
You may wish to set the price and amount here if the passed in `price` or `amount` is zero. The reason being is that they are optional parameters, and our exchange API implementations are not checking > 0 before sending to the API and one may modify an order unintentionally due to default values being `0`. You could place checks across all the exchange API level implementations of `ModifyOrder()`, but a check here also fixes the issue.